### DISCUSSION TITLE
large json object hang up chrome in tree mode

### DISCUSSION THREAD

**Discussion by yanshuai** [josdejong/svelte-jsoneditor#439] [Ideas]
I use svelte-jsoneditor in tree mode, for a json object having 40000 children k-v，it will hang up chrome, how can I limit  json tree that  show at most 1000 children default(if over 1000 children, then show click button to loading next 1000 children)

**Comment by josdejong** [comment]
For arrays, the editor only renders the first 100 items, and allows to expand other items. That keeps it fast for large arrays since the rendering is the heaviest part. For objects, there is not yet such an optimization because it isn't a common use case. It would be great of course to implement the same mechanism for objects.

I'll move this idea to the ideas section.

**Comment by josdejong** [comment]
An other idea based on a suggestion by Alex in #527: implement a virtual scroll like `table` mode has. Note that this is relatively complex to implement, since `tree` mode has a recursive, nested structure where every item can have a different height (which can be determined only after rendering).
